File: supermarket_flutter_app/lib/core/models/expense.dart
class Expense {
  final int id;
  final String description;
  final double amount;
  final DateTime date;
  final String category;
  final String? createdBy;

  Expense({
    required this.id,
    required this.description,
    required this.amount,
    required this.date,
    required this.category,
    this.createdBy,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      description: json['description'],
      amount: (json['amount'] as num).toDouble(),
      date: DateTime.parse(json['date']),
      category: json['category'],
      createdBy: json['createdBy'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'description': description,
      'amount': amount,
      'date': date.toIso8601String(),
      'category': category,
      'createdBy': createdBy,
    };
  }
}

File: SupermarketAPI/Models/Expense.cs
using System.ComponentModel.DataAnnotations;

namespace SupermarketAPI.Models
{
    public class Expense
    {
        public int Id { get; set; }

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Amount must be greater than zero")]
        public decimal Amount { get; set; }

        public DateTime Date { get; set; } = DateTime.UtcNow;

        [Required]
        public string Category { get; set; } = string.Empty; // e.g., "Utility", "Rent", "Salary", "Maintenance", "Other"

        public string? CreatedBy { get; set; } // Username or Employee ID
    }
}
